xarray/backends/zarr.py

```diff
@@ -307,6 +318,7 @@ def encode_zarr_variable(var, needs_copy=True, name=None):
     out : Variable
         A variable which has been encoded as described above.
     """
+    original_chunks = var.chunks
```

Comment:
This is kind of the crux. I cannot actually understand how / where the `.chunks` attributes is defined for Variables. We want the decoding pipeline to preserve chunks unmodified.

`.chunks` is not defined in the Variables class itself so must be somehow inherited from the other classes:
https://github.com/pydata/xarray/blob/15328b64a44ddb18c1c35c3126b58498a63e3dd5/xarray/core/variable.py#L315

The word `chunks` does not even appear in https://github.com/pydata/xarray/blob/main/xarray/coding/times.py.

I'm tempted to loop in @TomNicholas into this conversation, who recently refactored everything about how we handle chunked arrays, to help us sort through this.

Reply:
`chunks` is inherited from NamedArray:
https://github.com/pydata/xarray/blob/e5d163a8ddef4e8aa95a2841505a642f8ffbcea6/xarray/namedarray/core.py#L660

Yes, the problem is that the encoder does not make any difference between dask/numpy arrays and always returns numpy arrays. I originally thought that was a mistake, but I wasn't so sure after I tried to change that.

Reply:
DateTimes are always cast to numpy using `np.asarray` AFAICT. And this is there since that part of the coding was implemented.